### PR TITLE
Save off media transformations

### LINKED ISSUE #826: saving with media transformation config
MMM with `adstock` and `saturation` configurations can potentially not be saved correctly since the additional kwargs are not saved off. 

 Having these classes serializable and taking advantage of them in `save` and `load` methods would be a good idea.

 For instance, a model that would not be able to be loaded correctly would be:

 ```python
 from pymc_marketing.mmm import MMM, GeometricAdstock, LogisticSaturation

 mmm = MMM(
     ..., 
     adstock=GeometricAdstock(l_max=4, normalize=False, mode="Before"), 
     saturation=LogisticSaturation(),
     ..., 
)
```

This is because both the `normalize` and `mode` are not saved off.

This would involve reconstructing the `adstock` and `saturation` upon loading here: 

https://github.com/pymc-labs/pymc-marketing/blob/2be2664557fa84471e448d9ba12fe78ccacd69f3/pymc_marketing/mmm/delayed_saturated_mmm.py#L637-L659

### PR DESCRIPTION
**Pull request by wd60622**
## Description

Add `to_dict` method for media transformation and leverage in the `MMM` in order to save off configurations completely

## Related Issue

- [ ] Closes #826 
- [ ] Related to #

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [x] MMM
- [ ] CLV

## Type of change

- [ ] New feature / enhancement
- [x] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--882.org.readthedocs.build/en/882/

### CODE REVIEW

**Review by juanitorduz — APPROVED** [review]
Thanks @wd60622

**Comment by juanitorduz** [comment]
@wd60622 we can merge this one after we you solve the merge conflicts :)

File: tests/mmm/test_delayed_saturated_mmm.py
Comment (wd60622) [inline tests/mmm/test_delayed_saturated_mmm.py:1]: Any thoughts on this change? @juanitorduz 

If the adstock and saturation keys are not present than we *could* inject a default. However, I am not sure that is good idea

**Comment by wd60622** [comment]
Any idea if the model `version` attribute needs to be updated?

File: tests/mmm/test_delayed_saturated_mmm.py
Comment (juanitorduz) [reply tests/mmm/test_delayed_saturated_mmm.py:1]: Well, I think we need this to recover the model after saving it, right? What could be the drawbacks of injecting the default?

**Comment by juanitorduz** [comment]
> Any idea if the model `version` attribute needs to be updated?

We actually do not have any explicit convention. I usually do it when there is a big refactor, but we can change that.

File: tests/mmm/test_delayed_saturated_mmm.py
Comment (wd60622) [reply tests/mmm/test_delayed_saturated_mmm.py:1]: Injecting a default would likely not recreate the same model at all

**Pull request merged**
